src/modules/dna-service/dna.background.ts

```diff
@@ -248,7 +253,45 @@ export class DnaService {
     return browserStorage.set(ACTION_QUEUE_KEY, []);
   }
 
+  async registerAllWallets() {
+    const isInvoked = await browserStorage.get<boolean>(
+      REGISTER_ALL_WALLETS_INVOKED_KEY
+    );
+    if (isInvoked) {
+      return;
+    }
+
+    const wallet = this.getWallet();
+    const walletGroups = await wallet.uiGetWalletGroups({
+      context: INTERNAL_SYMBOL_CONTEXT,
+    });
+
+    const ownedAddresses =
+      walletGroups
+        ?.filter((group) => !isReadonlyContainer(group.walletContainer))
+        ?.flatMap((group) =>
+          group.walletContainer.wallets.map((wallet) => ({
+            address: wallet.address,
+            origin: group.origin || undefined,
+          }))
+        ) || [];
+
+    for (const { address, origin } of ownedAddresses) {
+      try {
+        await this.registerWallet({
+          params: { address, origin },
+        });
+      } catch (error) {
+        console.log('registerWallet error', error); // eslint-disable-line no-console
```

Comment:
As I see, we will try register each wallet only once. So if something goes wrong, we will not try again.
May be we should not mark register process as complete in this case and try again next time?

Reply:
I'll update it not set the flag if at least one wallet fails to register